REQUIRE

```diff
@@ -4,7 +4,7 @@ Distributions
 Docile
 DualNumbers
 FITSIO
-ForwardDiff 0.0.3 0.1
+ForwardDiff # Currently only required for testing.
```

Comment:
Hi! Sorry to drop in so randomly, but I just recently saw this package (neat stuff!) and happened to see these comments while doing some exploring. 

> Unfortunately, ForwardDiff types allocate a ton of memory

Unless there was a regression we didn't catch, ForwardDiff's number types should be laid out in memory in almost the exact same fashion as `Dual`s, as long as you configure them correctly (specifically `Dual{T}` ~ `GradientNumber{1,T,Tuple{T}}`). When you did your benchmark, what types did you actually use (i.e. what is the type of `mp_fd.vp`/is that code available somewhere)? 

While the point of ForwardDiffNumbers is to allow simultaneous accumulation of multiple derivatives per function call, another lesser-known feature is that the underlying storage type is swappable. You can use either `Vector` or `Tuple` types, which have different performance characteristics (this is exposed in the API via [the chunk_size keyword](http://www.juliadiff.org/ForwardDiff.jl/chunk_vec_modes.html), and to `C` in `ForwardDiffNumber{N,T,C}`). In many cases, using `Vector` storage can trigger GC, which will kill performance. I'd bet that's what's happening in your loop above.

Anyway, I don't mean to intrude - just thought this was a cool package and was curious about the issue you mentioned. I'm also curious about your use case of nesting `Dual`s, but I don't want to take up too much of your time!